### PR TITLE
Setup clang-format, tslint and editorconfig

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

File: src/lib/repeat.ts [hunk 1]
```diff
@@ -40,14 +43,16 @@ export function repeat<T>(items: Iterable<T>, keyFnOrTemplate: KeyFn<T>|ItemTemp
     let state = stateCache.get(part);
     if (state === undefined) {
       state = {
-        keyMap: keyFn && new Map(),
-        parts: [],
+        keyMap : keyFn && new Map(),
```
Comment (stramel) [inline src/lib/repeat.ts]: This is interesting, don't think I have seen anyone use this style before..

Comment (gdi2290) [reply src/lib/repeat.ts]: same, would prefer no space between the colon

File: src/lit-html.ts [hunk 2]
```diff
@@ -394,17 +395,20 @@ export class NodePart implements SinglePart {
...
-  } else if (templatePart.type === 'node') {
-    return new NodePart(instance, node, node.nextSibling!);
-  }
-  throw new Error(`Unknown part type ${templatePart.type}`);
-}
+export type PartCallback = (instance: TemplateInstance,
+                            templatePart: TemplatePart, node: Node) => Part;
+
+export const defaultPartCallback =
+    (instance: TemplateInstance, templatePart: TemplatePart, node: Node):
```
Comment (gdi2290) [inline src/lit-html.ts]: not sure how this is better lol

Comment (stramel) [reply src/lit-html.ts]: I'm guessing it was to fix line length? /shrug

File: package.json [hunk 3]
```diff
@@ -13,16 +13,19 @@
     "build": "tsc",
     "gen-docs": "typedoc --readme none --mode modules --excludeNotExported --excludePrivate --exclude **/*_test.ts --out ./docs/api .",
     "pretest": "npm run posttest; ln -s node_modules bower_components",
-    "test": "wct -l chrome",
+    "test": "npm run lint && wct -l chrome",
     "posttest": "rm -f bower_components",
-    "checksize": "uglifyjs lit-html.js -mc --toplevel | gzip -9 | wc -c"
+    "checksize": "uglifyjs lit-html.js -mc --toplevel | gzip -9 | wc -c",
+    "format": "find src test | grep '\\.js$\\|\\.ts$' | xargs clang-format --style=file -i",
```
Comment (rictic) [inline package.json]: I think this PR is missing the .clang-format file

File: package.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
     "build": "tsc",
     "gen-docs": "typedoc --readme none --mode modules --excludeNotExported --excludePrivate --exclude **/*_test.ts --out ./docs/api .",
     "pretest": "npm run posttest; ln -s node_modules bower_components",
-    "test": "wct -l chrome",
+    "test": "npm run lint && wct -l chrome",
```
Comment (rictic) [inline package.json]: I'd recommend running lint after tests. Most of the time your editor will tell you about lint errors as you work.

I would recommend building before testing though. Something like:

```
"test": "npm run build && wct -l chrom && npm run lint",
```

File: src/lib/repeat.ts (hunk 1 above)
Comment (justinfagnani) [reply src/lib/repeat.ts]: Same, but every formatter I've tried - prettier, clang and tsfmt - have their warts. We can file bugs on clang.

File: src/lit-html.ts (hunk 2 above)
Comment (justinfagnani) [reply src/lit-html.ts]: yep. You take the bad with the consistency

File: package.json (hunk 4 above)
Comment (justinfagnani) [reply package.json]: done. fyi a lot of our projects have the previous test command

File: package.json (hunk 3 above)
Comment (justinfagnani) [reply package.json]: Ahhh, much better!

File: src/lib/repeat.ts (hunk 1 above)
Comment (justinfagnani) [reply src/lib/repeat.ts]: fixed with a .clang-format

**Comment by justinfagnani** [comment]
@rictic PTAL

**Pull request merged**